apps/web/src/app/_components/reviews/new-role-dialogue.tsx

```diff
@@ -0,0 +1,147 @@
+import { useState } from "react";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useForm } from "react-hook-form";
+import { z } from "zod";
+
+import { Button } from "@cooper/ui/button";
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from "@cooper/ui/dialog";
+import { Form, FormControl, FormField, FormItem } from "@cooper/ui/form";
+import { useToast } from "@cooper/ui/hooks/use-toast";
+import { Input } from "@cooper/ui/input";
+import { Label } from "@cooper/ui/label";
+import { Textarea } from "@cooper/ui/textarea";
+
+import { api } from "~/trpc/react";
+
+const roleSchema = z.object({
+  title: z.string({ required_error: "You need to enter a role title." }),
+  description: z.string(),
+  companyId: z.string(),
+});
+
+export type RoleRequestType = typeof roleSchema;
+
+interface NewRoleDialogProps {
+  companyId: string;
+}
+
+export default function NewRoleDialog({ companyId }: NewRoleDialogProps) {
+  const [isSuccess, setIsSuccess] = useState(false);
+  const company = api.company.getById.useQuery({ id: companyId });
+  const [roleName, setRoleName] = useState("");
+  const form = useForm<z.infer<RoleRequestType>>({
+    resolver: zodResolver(roleSchema),
+    defaultValues: {
+      title: "",
+      description: "",
+      companyId: companyId,
+    },
+  });
+
+  const { toast } = useToast();
+
+  const mutation = api.role.create.useMutation({
+    onSuccess: () => {
+      setIsSuccess(true);
+      setRoleName(form.getValues().title);
+
+      setTimeout(() => {
+        window.location.reload();
+      }, 1000);
+    },
+    onError: (error) => {
+      toast({
+        title: "Submission Error",
+        description: error.message || "Something went wrong. Please try again.",
+        variant: "destructive",
+      });
+    },
+  });
+
+  async function onSubmit(values: z.infer<RoleRequestType>) {
+    try {
+      await mutation.mutateAsync(values);
+    } catch (error) {
+      console.error(error);
+    }
+  }
+
+  return (
+    <Dialog>
+      <DialogTrigger asChild>
+        <Button className="h-9 rounded-full border-none border-cooper-yellow-500 bg-cooper-yellow-500 px-4 py-3 text-sm font-semibold text-white hover:border-cooper-yellow-300 hover:bg-cooper-yellow-300">
+          + Create New Role
+        </Button>
+      </DialogTrigger>
+
+      <DialogContent className="w-full bg-white">
+        {isSuccess ? (
+          <div className="flex flex-col items-center justify-center gap-4 py-8">
+            <h2 className="text-xl font-semibold text-cooper-green-500">
+              Success!
+            </h2>
+            <p className="text-center text-gray-600">
+              The {roleName} co-op has been successfully added to{" "}
```

Comment:
This is an extreme nitpick but if the submit handler, maybe we can check if the word "co-op" is in the string and if it is we remove it? Idk whatever you think LOL!